### PR TITLE
Uninversal aws infrastructure

### PR DESCRIPTION
**Pull request by Sammers21**
Add universal aws infrastructure with a purpose to be used in every benchmark

Related to #21

### CODE REVIEW

**Comment by Sammers21** [comment]
@g4s8, can you review this PR?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@Sammers21 why to create new terraform script for AWS infrastructure? We already have working scripts and you can modify it. New implementation doesn't have few important details, such as security groups to access instances via SSH, VPC configuration to setup private network for these instances. Also, we have a script for waiting instances boot for complete https://github.com/artipie/benchmarks/blob/master/files/wait-instance-init.sh

**Comment by Sammers21** [comment]
@g4s8, review the PR once again,  please

**Comment by Sammers21** [comment]
@g4s8, please, review the PR.

File: aws-infrastructure/infrastructure.tf [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  tags = {
+    Name = "perf-net",
+    Project = "Artipie Performance"
+  }
+}
+
+resource "aws_subnet" "perf_subnet" {
+  cidr_block = cidrsubnet(aws_vpc.perf_net.cidr_block, 12, 1234)
+  vpc_id = aws_vpc.perf_net.id
+  availability_zone = "eu-central-1b"
```
Comment (g4s8) [inline aws-infrastructure/infrastructure.tf]: @Sammers21 let's use variable for AWS zone to always use same reference in all configurations

**Comment by Sammers21** [comment]
@g4s8, review the PR, please.

File: aws-infrastructure/infrastructure.tf [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    type = "ssh"
+    user = "ubuntu"
+    host = self.public_ip
+    private_key = file("aws_ssh_key")
+  }
+}
+
+resource "aws_instance" "server" {
+  ami = data.aws_ami.ubuntu.id
+  instance_type = "t2.medium"
```
Comment (g4s8) [inline aws-infrastructure/infrastructure.tf]: @Sammers21 let's move insance type to the variable too

File: aws-infrastructure/start.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Start AWS infrotructure
+terraform init
+terraform apply -input=false -auto-approve
+
+# Export terraform output variables
+export PRIVATE_CLIENT_IP_ADDR=$(terraform output private_client_ip_addr)
+export PRIVATE_SERVER_IP_ADDR=$(terraform output private_server_ip_addr)
+export PUBLIC_CLIENT_IP_ADDR=$(terraform output public_client_ip_addr)
+export PUBLIC_SERVER_IP_ADDR=$(terraform output public_server_ip_addr)
+
```
Comment (g4s8) [inline aws-infrastructure/start.sh]: @Sammers21 why do we need these env variables? It'll have script's scope of visibility and it'll not be possible to access them outside the script

**Review by g4s8 — CHANGES_REQUESTED** [review]
@Sammers21 please check my comments

File: aws-infrastructure/start.sh (hunk 3 above)
Comment (Sammers21) [reply aws-infrastructure/start.sh]: @g4s8, for SSH connection checking i guess

**Comment by Sammers21** [comment]
@g4s8, review the PR once again, please.

**Review by g4s8 — APPROVED** [review]
@Sammers21 thanks

**Pull request merged**